src/win/pipe.c

```diff
@@ -985,6 +1079,8 @@ static DWORD WINAPI uv_pipe_zero_readfile_thread_proc(void* parameter) {
           uv_mutex_lock(m);
           handle->pipe.conn.readfile_thread = hThread;
           uv_mutex_unlock(m);
+          /* Give some time for other processes to wake before restarting. */
+          Sleep(1);
```

Comment:
Unfortunately - yes. Without it the test won't work. Interrupter thread will be stuck at [CancelSynchronousIo](https://github.com/libuv/libuv/pull/1273/files#diff-7be27b71e7f7a6d2c40c447747db1231R968), zero-read at [ReadFile](https://github.com/libuv/libuv/pull/1273/files#diff-7be27b71e7f7a6d2c40c447747db1231R1066), and the test thread at [NtQueryInformationFile](https://github.com/libuv/libuv/pull/1273/files#diff-7be27b71e7f7a6d2c40c447747db1231R2037). For whatever reason zero-read thread seems to ignore the `CancelSynchronousIo` call, unless a `Sleep` (or a `printf`, or a attached debugger) is present.